lnwallet/channel.go

```diff
@@ -3892,16 +3896,41 @@ func (lc *LightningChannel) ProcessChanSyncMsg(
 			return nil, nil, nil, err
 		}
 
+		// If lastWasRevoke is set to true, a revocation was last and we
+		// need to reorder the updates.
+		var lastWasRevoke bool
+
+		// We check the length of the updates slice to check if we already
+		// determined that we owe a revocation.
+		if len(updates) != 0 {
+			// If a revocation is owed, then we need to determine the proper
+			// ordering for updates. In the case that ErrLastNotFound is
+			// returned and the last update is not stored, we do nothing and
+			// follow the original ordering.
+			lastWasRevoke, err = lc.channelState.LastWasRevoke()
+			if err != nil && err != channeldb.ErrLastNotFound {
+				return nil, nil, nil, err
+			}
+		}
+
+		var commitUpdates []lnwire.Message
+
 		// Next, we'll need to send over any updates we sent as part of
 		// this new proposed commitment state.
 		for _, logUpdate := range commitDiff.LogUpdates {
-			updates = append(updates, logUpdate.UpdateMsg)
+			commitUpdates = append(commitUpdates, logUpdate.UpdateMsg)
 		}
 
 		// With the batch of updates accumulated, we'll now re-send the
 		// original CommitSig message required to re-sync their remote
 		// commitment chain with our local version of their chain.
-		updates = append(updates, commitDiff.CommitSig)
+		commitUpdates = append(commitUpdates, commitDiff.CommitSig)
+
+		if lastWasRevoke {
```

Comment:
a comment here would be useful to explain how these two orderings differ

Reply:
Comment added